torch/distributed/algorithms/join.py

```diff
@@ -43,8 +44,10 @@ def post_hook(self, is_last_joiner: bool):
 class _Joinable(ABC):
     r"""
     This defines an abstract base class for joinable classes. A joinable class
-    (inheriting from :class:`_Joinable`) should implement a private
```

Comment:
Is _Joinable ever intended to be used by non PyTorch developers? If so we should eventually remove the _ prefix and make it a public API.

Reply:
Yup, I think it is intended to be used by non-PyTorch developers eventually. Everything (meaning `_Joinable`, `_Join`, `_JoinHook` and also `DistributedDataParallel._join_hook()`, `ZeroRedundancyOptimizer._join_hook()`) has a prefix underscore for now. I am not sure when would be the time to do the removal, and I was originally waiting for the feature to be approved. I also think it would be good to discuss exactly which components will be made public and which are kept private.